### DISCUSSION TITLE
🤏🏻 Use `investiny` while we fix `investpy`!

### DISCUSSION THREAD

**Discussion by alvarobartt** [alvarobartt/investpy#612] [General]
Hi everyone, I'm Alvaro Bartolome, ML Engineer, open source developer, and the creator of investpy. As many of you may already know, Investing.com which is the data source that investpy uses have recently changed its APIs as well as the protection of those with Cloudflare, which means that the data is not accessible the same way as before. I and other community members are working actively on solving this issue and looking for alternatives to pull the data from Investing.com.

Most of the solutions proposed by the community worked at some point, but are now discontinued or no longer working, but one solution proposed by @ramakrishnamekala129 at https://github.com/alvarobartt/investpy/issues/610 seems to be working fine and consistently.

So on, I've decided to develop investiny in the meantime, so that everyone can pull the data from Investing.com but in a simpler and faster way. Also as Investing.com is changing so many things lately, I decided to create a new repository named investiny so as to be easily adaptative to changes. Note that investiny is not a replacement for investpy, just an intermediate solution in the meantime while we solve the current issues with investpy.

I'd like all of you to star investiny and contribute to it, as the code is much simpler and easy to understand, but it obviously has less functionality, but we'll get there at some point. Also showing me support by starring, liking, retweeting, etc. is much appreciated.

Thanks, everyone! 🤗

**Comment by alvarobartt** [comment]
## ❓ How to use it?

```python
from investiny import historical)data

data = historical_data(investing_id=6408, from_date="09/01/2022", to_date="10/01/2022")
print(data)
```

## 🤔 Q&A

* **Where does the `investing_id` parameter come from?** -> It's the Investing.com unique identifier for each asset that they have listed in their platform, you can look for it either by inspecting the code of Investing.com, or just checking the ID field in the `investpy/resources` which are a collection of CSV files scraped from Investing.com and contain some of that inner information.

For example, the value `investing_id=6408` used in the example above, is retrieved from `united states,Apple,Apple Inc,apple-computer-inc,US0378331005,6408,USD,AAPL` in https://raw.githubusercontent.com/alvarobartt/investpy/master/investpy/resources/stocks.csv from the column `id`.
* **Why `investiny` instead of publishing a patch to `investpy`?** -> Since Investing.com is constantly changing things, and currently the `investpy` code is outdated and not well formatted, I've decided to create `investiny` to easily develop new features there and let everyone contribute as the code is much simpler, while also improving the size of the package, the requests' latency, etc.

`investiny` available at https://github.com/alvarobartt/investiny

**Reply by plutoBase** [reply]
Thanks for all your hard work, Alvaro, much appreciated!

I am having trouble finding some securities I have been using recently in the CSV files in the resources section. They simply don't seem to be present!  Would it be possible to either compile a new list, or alternatively can you give me a clue where to look for the id tag in the page source (ie an html tag reerence)??

Many thanks!

**Reply by plutoBase** [reply]
never mind, Alvaro, I actually managed to engage my brain!

Thanks again

**Reply by alvarobartt** [reply]
Hi @plutoBase in the latest `investiny` release I introduced the `search_assets` function so that you can combine both `search_assets` and `historical_data`, without knowing the Investing.con ID or looking for it manually, you can just retrieve it from `search_assets` results as detailed in https://alvarobartt.github.io/investiny/usage 🤗

**Reply by rnayaksec** [reply]
Is there an equivalent for investpy.etfs.get_etf_information() to be able to pull basic information about a set of ETFs?

**Comment by gowrilokeshisi** [comment]
Hi,
I want to know that can we use investpy for wealth management/advisory at a commercial level. Do i require to take a licence for the same. Please guide me on the process.
Thanks,
Lokesh

**Reply by alvarobartt** [reply]
Hi @gowrilokeshisi as you can see in the `README.md`, `investpy` is just intended to be used for research and/or educational purposes, so it cannot be used for commercial purposes, as it's using data from an external source (Investing.com) and `investpy` is not related at all with Investing.com or have any relationship. Anyway, feel free to contact Investing.com in case you want to use their services and/or data :smile:

**Comment by mk1991** [comment]
Hi, 

I am getting 403 error when requesting historical data using investiny.

Any updates on how this is going, or on the fix for investpy?

**Comment by PrinceMalaba** [comment]
I am also using investiny but I am still getting connection error 403. Kindly assist I want to retrieve stock historical data.

**Reply by ymyke** [reply]
Both investpy and investiny no longer work bc investing.com prevents scraping. I recommend using https://github.com/ymyke/tessa , which uses yfinance and pycoingecko.
